frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -562,6 +563,17 @@ export const insightLogic = kea<insightLogicType>({
                 actions.loadResults()
             }
         },
+        reportInsightViewedForRecentInsights: async () => {
+            // Report the insight being viewed to our '/viewed' endpoint. Used for "recently viewed insights"
+
+            // NOTE: This should be merged into the same action as `reportInsightViewed`, but we can't right now
+            // because there are some issues with `reportInsightViewed` not being called when the
+            // insightLogic is already loaded.
+            // For example, if the user navigates to an insight after viewing it on a dashboard, `reportInsightViewed`
+            // will not be called. This should be fixed when we refactor insightLogic, but the logic is a bit tangled
+            // right now
+            api.create(`api/projects/${teamLogic.values.currentTeamId}/insights/${values.insight.id}/viewed`)
+        },
```

Comment:
I think I understand what you're trying to do here, but perhaps a TODO here could help signal that this should be removed once the existing `reportInsightViewed` bug is fixed.